cfpb/sbl-frontend
# Type of Financial Institution (api integration)

Closes #291 

## Changes

- Save `Type of Financial institution` form data via API
- Creates separate TS type for the form (needed for form validation)
- Expand `<FormMain>` to accept an `onSubmit` function, which allows us to intercept/validate when a user hits "Enter"
- Fixes some issues when linking to the next incomplete step (Warnings and Sign/submit were not being shown)

## Notes
- The testing labels will be removed as part of Filing home updates:
  - Start the filing process *(needs type of fi)*
  - Start the filing process *(create filing)*
 


## How to test this PR

1. Start clean (compose down -v and compose up)
1. Create at least one institution *without* any associated SBL institution types.  I do this via the seed script by editing the JSON of an institution to have `"sbl_institution_types": []`.  
2. Associate user with institutions
3. On the Filing home page you should see at least one institution card with the heading `Start the filing process (needs type of fi)`.  
4. Clicking `Start filing` for this institution should take you to the `Type of Financial institution` form
5. Form validations should prevent you from continuing until you've correctly filled out the form
6. Upon form submission, you will be routed to the `Upload` page
7. Return to the Filing home page (`/filing`)
8. The institution you just updated should now show that the next step is `Upload file`
7. Return to the Filing home page (`/filing`)
10. For an institution that already has associated types but has not started the filing process, you should see the heading `Start the filing process (create filing)`.  
11. Clicking `Start filing` for this institution should route you to the `Upload` page
7. Return to the Filing home page (`/filing`)
8. The institution you just started the filing process for should now show that the next step is `Upload file`
9. Done

## Screenshots
### Flow: Institution that needs to fill out SBL Type of FI form
https://github.com/cfpb/sbl-frontend/assets/2592907/a7efbd6f-f38f-46e4-8448-df6e2a33506d

### Flow: Institution that already has SBL Type of FI
https://github.com/cfpb/sbl-frontend/assets/2592907/7fc7d02a-0a30-43c5-88c2-9f3809934c95
